pipeline_dp/combiners.py

```diff
@@ -175,16 +175,49 @@ def additive_vector_noise_params(
             noise_kind=self.aggregate_params.noise_kind)
 
 
-class CountCombiner(Combiner):
+class MechanismContainerMixin(abc.ABC):
+    """Abstract class with implementation of handling DP mechanism."""
+
+    @abc.abstractmethod
+    def sensitivities(self) -> dp_computations.Sensitivities:
+        pass
+
+    @abc.abstractmethod
+    def mechanism_spec(self) -> budget_accounting.MechanismSpec:
+        pass
+
+    def __getstate__(self):
```

Comment:
What is state? I don't see references to it in the rest of the code.

Reply:
Python methods \_\_some_name\._\_ are service methods (like \_\init\_\_ \_\eq\._\_ etc).
\_\getstate\._\_ is called when serialization happens, [some examples](https://stackoverflow.com/questions/1939058/simple-example-of-use-of-setstate-and-getstate).

I've aded comment in this function to explain in more details.